src/back/TicketManager/src/PaymentService/TicketManager.PaymentService.Services/ApiKeyAuth/ApiKeyAuthorization.cs

```diff
@@ -0,0 +1,31 @@
+using FastEndpoints;
+using FluentValidation.Results;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using TicketManager.PaymentService.Services.Configuration;
+
+namespace TicketManager.PaymentService.Services.ApiKeyAuth;
+
+public class ApiKeyAuthorization<TRequest> : IPreProcessor<TRequest>
+{
+    private const string ApiKeyHeaderName = "pay_api_key";
+    private PaymentServiceConfiguration configuration; 
+    
+    public ApiKeyAuthorization(PaymentServiceConfiguration configuration) {
+        this.configuration = configuration;
+    }
+    public Task PreProcessAsync(TRequest req, HttpContext context, List<ValidationFailure> failures, CancellationToken ct)
+    {
+        if (!context.Request.Headers.TryGetValue(ApiKeyHeaderName, out
+                var extractedApiKey)) {
+            context.Response.StatusCode = 401;
+            return context.Response.SendForbiddenAsync(ct);
```

Comment:
Here you return status code 403, not 401.

Reply:
Address in bac63fe0.